src/bindings/js/node/src/async_infer_queue.cpp

```diff
@@ -110,7 +113,10 @@ void AsyncInferQueue::set_custom_callbacks(const Napi::CallbackInfo& info) {
                         Napi::Object js_ir = InferRequestWrap::wrap(env, m_requests[handle]);
                         const auto promise = m_user_ids[handle].second;
                         try {
-                            user_callback.Call({js_ir, m_user_ids[handle].first.Value()});
+                            auto user_data = m_user_ids[handle].first.Value().ToString().Utf8Value() == "NULL"
```

Comment:
```suggestion
                            // We use this workaround because an ObjectReference cannot contain null or undefined.
                            auto user_data = m_user_ids[handle].first.Value().ToString().Utf8Value() == "NULL"
```